dll/src/ReflectiveLoader.c

```diff
@@ -455,15 +455,34 @@ RDIDLLEXPORT ULONG_PTR WINAPI ReflectiveLoader(VOID)
 {
 	LOADER_CONTEXT context = {0};
 
-	// Initialize the Syscall structures in our context. We use C99 designated initializers
-	// for clarity and to ensure the correct fields are populated.
-	// The remaining fields (dwSyscallNr, pStub) are zero-initialized by the LOADER_CONTEXT
-	// initialization and will be populated later by getSyscalls().
-	context.Syscalls[SyscallIndexAllocateVirtualMemory] = (Syscall){.dwCryptedHash = ZWALLOCATEVIRTUALMEMORY_HASH, .dwNumberOfArgs = 6};
-	context.Syscalls[SyscallIndexProtectVirtualMemory] = (Syscall){.dwCryptedHash = ZWPROTECTVIRTUALMEMORY_HASH, .dwNumberOfArgs = 5};
-	context.Syscalls[SyscallIndexFlushInstructionCache] = (Syscall){.dwCryptedHash = ZWFLUSHINSTRUCTIONCACHE_HASH, .dwNumberOfArgs = 3};
+	// Initialize the Syscall structures in our context.
+	// We use explicit member assignment instead of C99 designated initializers
+	// to maintain compatibility with C++ compilers (pre-C++20), which do not
+	// support this feature. The entire struct is first zero-initialized.
+	context.Syscalls[SyscallIndexAllocateVirtualMemory] = (Syscall){0};

```

Comment:
```suggestion
	context.Syscalls[SyscallIndexAllocateVirtualMemory] = {0};
```
I'm sorry, for the back and forth